### PR TITLE
BXMSPROD-401: fix slave image build job DSL

### PR DESCRIPTION
**Pull request by winklerm**
Updated slave image build job DSL to avoid authentication issues when cloning from Gerrit.

### CODE REVIEW

**Comment by winklerm** [comment]
@mareknovotny @mbiarnes @AntStephenson FYI

**Comment by winklerm** [comment]
Please do not merge yet, I would like to finish testing the image before (and include possible changes here).

**Comment by winklerm** [comment]
I have just checked that both images have the RAW format which makes both of them 40 GB, so the size should not make a difference.

**Comment by winklerm** [comment]
One final image build test job run and we will be ready for merging.

**Comment by winklerm** [comment]
There are some issues with the image build job running after freshly generated from the job DSL, investigating.

File: job-dsls/jobs/slave_image_build.groovy [hunk 1]
```diff
@@ -52,7 +74,7 @@ def jobDefinition = job("slave-image-build") {
             string("PSI_PASSWORD", "psi-rhba-jenkins-password")
 
             // Copies the file given in the credentials to a temporary location, then sets the variable to that location.
-            file("PSI_PRIVATE_KEY", "kie-jenkins.pem")
+            file("PSI_PRIVATE_KEY", "kie-jenkis.pem")
```
Comment (mareknovotny) [inline job-dsls/jobs/slave_image_build.groovy]: typo?

**Comment by winklerm** [comment]
Issues fixed by adding another provisioner for packer clearing up the trouble-making temp files. Ready to review/merge.

File: job-dsls/jobs/slave_image_build.groovy (hunk 1 above)
Comment (winklerm) [reply job-dsls/jobs/slave_image_build.groovy]: Unfortunately, this typo is in the Jenkins master configuration and I cannot rename it there (it is typo in credential ID). I would need to recreate the credential from scratch, but I need to get out the .pem file first, it cannot be easily retrieved from Jenkins UI.

**Comment by winklerm** [comment]
I do not understand why I cannot remove the DO NOT MERGE label.. probably because I am missing some rights?

**Pull request merged**